pkg/instance/instance.go

```diff
@@ -447,25 +478,48 @@ func (s *Server) watchSPDKReplica(ctx context.Context, req *emptypb.Empty, srv r
 		return errors.Wrap(err, "failed to create SPDK replica watch notifier")
 	}
 
+	failureCount := 0
 	for {
+		if failureCount >= maxMonitorRetryCount {
+			logrus.Errorf("Continuously receiving errors for %v times, stopping watching SPDK replicas", maxMonitorRetryCount)
+			return fmt.Errorf("continuously receiving errors for %v times, stopping watching SPDK replicas", maxMonitorRetryCount)
+		}
+
 		select {
 		case <-ctx.Done():
-			logrus.Info("Stopped watching SPDK replica")
-			return nil
+			logrus.Info("Stopped watching SPDK replicas")
+			return ctx.Err()
 		default:
-			_, err := notifier.Recv()
-			if err != nil {
-				logrus.WithError(err).Error("Failed to receive next item in SPDK replica watch")
-			} else {
-				if err := srv.Send(&empty.Empty{}); err != nil {
-					return errors.Wrap(err, "failed to send instance response")
+			// Use a channel to receive the result of notifier.Recv()
+			errChan := make(chan error, 1)
+
+			go func() {
+				_, err := notifier.Recv()
+				errChan <- err
+				close(errChan)
+			}()
+
+			select {
+			case <-ctx.Done():
+				// If context is done before notifier.Recv() completes, return immediately
+				logrus.Info("Stopped watching SPDK replicas")
+				return ctx.Err()
```

Comment:
This `select` looks duplicated...

Reply:
Actually, I am not sure why you prefer this kind of implementation. Are you worried that the potential blocking caused by `notifier.Recv()` makes `<-ctx.Done()` not be handled in time?

Reply:
You're right.
I'm thinking which one of `go func() {...}` and `ticker method you implemented in go-spdk-helper` is better?

Reply:
Using another solution, please help check it. Thank you.